sematic/resolvers/log_streamer.py

```diff
@@ -156,13 +238,34 @@ def ingested_logs(
         # it will go to stdout before stdout gets redirected.
         print(
             f"To follow these logs, try:\n\t"
-            f"kubectl exec -i {pod_name} -- tail -f {file_path}"
+            f"kubectl exec -i {pod_name} -- tail {file_path}"
         )
-    final_upload_error = None
+
+    original_signal_handler = None
+    streamer_pid = None
+
+    def clean_up_streamer(signal_num, frame=None):
+        if streamer_pid is not None:
+            # forwarding the signal should trigger a final upload.
+            # use a timeout so the parent process can still exit if
+            # the child hangs for some reason (ex: during remote service call)
+            _send_signal_or_kill(streamer_pid, signal_num, timeout_seconds=20)
+        if original_signal_handler is not None and hasattr(
```

Comment:
Airing out for readability.
```suggestion

        if original_signal_handler is not None and hasattr(
```